Azure/azure-cli
# [RBAC] ad sp delete: silence 'application not found' error

Fix #10493: `az ad sp delete --id {}` fails when application is not found. If not found, application deletion is skipped.

To test:
```
az login -t {tenant1}
az ad app create --display-name {} --available-to-other-tenants

az login -t {tenant2}
az ad sp create --id {appId}
az ad sp delete --id {appId} --verbose
```

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).
